.github/workflows/publish.yaml

```
@@ -35,14 +35,26 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+
       - uses: actions/setup-python@v4
         with:
           python-version: "3.11"
 
+      - uses: actions/setup-node@v3
+        with:
+          cache: yarn
+          node-version: 18.x
+          registry-url: https://registry.npmjs.org
+          cache-dependency-path: labextension/yarn.lock
+
+      - name: Update root build packages
+        run: |
+          pip install --upgrade build pip jupyterlab~=3.0
```

Is `jupyterlab~=3.0` needed to be specified here? I'm thinking that pyproject-build ought to handle this if its outlined as a requirement for the build env in pyproject.toml?

I tried and got this:

```
* Installing packages in isolated environment... (hatch-jupyter-builder >=0.5, hatch-nodejs-version, hatchling >=1.4.0, jupyterlab >=3.4.7,<4.0.0)
```

```suggestion
          pip install --upgrade build pip
```

---

Further down i see that we only have the following, so it seems correct to remove this part.

```
      - name: Update root build packages
        run: |
          pip install --upgrade build pip
      - name: Build Python package
        run: |
          pyproject-build
```

---

using `npm pack` (likely better, anyway, as then the as-built and as-uploaded SHA256 will always match... `yarn` does... different things)